src/osiris_log.erl

```diff
@@ -585,7 +585,7 @@ maybe_fix_corrupted_files(#{dir := Dir}) ->
     ok = maybe_fix_corrupted_files(sorted_index_files(Dir));
 maybe_fix_corrupted_files([IdxFile]) ->
     SegFile = segment_from_index_file(IdxFile),
-    ok = truncate_invalid_idx_records(IdxFile, file_size(SegFile)),
+    ok = truncate_invalid_idx_records(IdxFile, file_size_or_zero(SegFile)),
```

Comment:
only seen the issue with multiple index files. I'm not sure if it can occur with a single index file so I just added the safeguard for symmetry.

Reply:
I think it can happen if the system it terminated as a stream is declared. this is fine.